src/main/java/com/faforever/api/clan/ClansController.java

```diff
@@ -40,24 +43,25 @@ public ClansController(ClanService clanService, PlayerService playerService) {
       @ApiResponse(code = 200, message = "Success with JSON { player: {id: ?, login: ?}, clan: { id: ?, name: ?, tag: ?}}"),
       @ApiResponse(code = 400, message = "Bad Request")})
   @RequestMapping(path = "/me", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
-  public Map<String, Serializable> me(Authentication authentication) {
+  public MeResult me(Authentication authentication) {
     Player player = playerService.getPlayer(authentication);
 
-    ImmutableMap<String, Serializable> playerMap = ImmutableMap.of(
-        "id", player.getId(),
-        "login", player.getLogin());
+    MeResult result = new MeResult();
+
+    result.setPlayer(new PlayerResult()
+        .setId(player.getId())
+        .setLogin(player.getLogin()));
```

Comment:
In order to ensure that clients can use the same mechanism to deserialize any object, we need to stick to the JSON-API.

You can achieve this like so:
```
    return new JsonApiDocument(new Data<>(new Resource("player", String.valueOf(player.getId()),
        ImmutableMap.of(
            "login", player.getName()
        ), null, null, null)));
```
(and attach the clan below accordingly).

However, in this case I still think there is a way to directly return `/player/1234`. Otherwise the client may fail to deserialize because the format doesn't match, but the type is the same. If we can't do that, we need to use another type than `player`.